{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyBioGate import molecular_profiles as mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecularAlterationType</th>\n",
       "      <th>datatype</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>showProfileInAnalysisTab</th>\n",
       "      <th>patientLevel</th>\n",
       "      <th>molecularProfileId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>genericAssayType</th>\n",
       "      <th>pivotThreshold</th>\n",
       "      <th>sortOrder</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PROTEIN_LEVEL</td>\n",
       "      <td>LOG2-VALUE</td>\n",
       "      <td>Protein expression (RPPA)</td>\n",
       "      <td>Protein expression measured by reverse-phase p...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_rppa</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PROTEIN_LEVEL</td>\n",
       "      <td>Z-SCORE</td>\n",
       "      <td>Protein expression z-scores (RPPA)</td>\n",
       "      <td>Protein expression, measured by reverse-phase ...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_rppa_Zscores</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COPY_NUMBER_ALTERATION</td>\n",
       "      <td>DISCRETE</td>\n",
       "      <td>Putative copy-number alterations from GISTIC</td>\n",
       "      <td>Putative copy-number calls on 90 cases determi...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_gistic</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COPY_NUMBER_ALTERATION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>Capped relative linear copy-number values</td>\n",
       "      <td>Capped relative linear copy-number values for ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_linear_CNA</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MUTATION_EXTENDED</td>\n",
       "      <td>MAF</td>\n",
       "      <td>Mutations</td>\n",
       "      <td>Mutation data from whole exome sequencing. Mut...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_mutations</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecularAlterationType    datatype  \\\n",
       "0           PROTEIN_LEVEL  LOG2-VALUE   \n",
       "1           PROTEIN_LEVEL     Z-SCORE   \n",
       "2  COPY_NUMBER_ALTERATION    DISCRETE   \n",
       "3  COPY_NUMBER_ALTERATION  CONTINUOUS   \n",
       "4       MUTATION_EXTENDED         MAF   \n",
       "\n",
       "                                           name  \\\n",
       "0                     Protein expression (RPPA)   \n",
       "1            Protein expression z-scores (RPPA)   \n",
       "2  Putative copy-number alterations from GISTIC   \n",
       "3     Capped relative linear copy-number values   \n",
       "4                                     Mutations   \n",
       "\n",
       "                                         description  \\\n",
       "0  Protein expression measured by reverse-phase p...   \n",
       "1  Protein expression, measured by reverse-phase ...   \n",
       "2  Putative copy-number calls on 90 cases determi...   \n",
       "3  Capped relative linear copy-number values for ...   \n",
       "4  Mutation data from whole exome sequencing. Mut...   \n",
       "\n",
       "   showProfileInAnalysisTab  patientLevel     molecularProfileId   studyId  \\\n",
       "0                     False         False          acc_tcga_rppa  acc_tcga   \n",
       "1                      True         False  acc_tcga_rppa_Zscores  acc_tcga   \n",
       "2                      True         False        acc_tcga_gistic  acc_tcga   \n",
       "3                     False         False    acc_tcga_linear_CNA  acc_tcga   \n",
       "4                      True         False     acc_tcga_mutations  acc_tcga   \n",
       "\n",
       "  genericAssayType  pivotThreshold sortOrder  \n",
       "0              NaN             NaN       NaN  \n",
       "1              NaN             NaN       NaN  \n",
       "2              NaN             NaN       NaN  \n",
       "3              NaN             NaN       NaN  \n",
       "4              NaN             NaN       NaN  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = mf.get_all_molecular_profiles()\n",
    "df1.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecularAlterationType</th>\n",
       "      <th>genericAssayType</th>\n",
       "      <th>datatype</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>showProfileInAnalysisTab</th>\n",
       "      <th>patientLevel</th>\n",
       "      <th>molecularProfileId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>study_name</th>\n",
       "      <th>...</th>\n",
       "      <th>study_publicStudy</th>\n",
       "      <th>study_pmid</th>\n",
       "      <th>study_citation</th>\n",
       "      <th>study_groups</th>\n",
       "      <th>study_status</th>\n",
       "      <th>study_importDate</th>\n",
       "      <th>study_readPermission</th>\n",
       "      <th>study_studyId</th>\n",
       "      <th>study_cancerTypeId</th>\n",
       "      <th>study_referenceGenome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GENERIC_ASSAY</td>\n",
       "      <td>ARMLEVEL_CNA</td>\n",
       "      <td>CATEGORICAL</td>\n",
       "      <td>Putative arm-level copy-number from GISTIC</td>\n",
       "      <td>Putative arm-level copy-number from GISTIC 2.0.</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>gbm_tcga_pan_can_atlas_2018_armlevel_cna</td>\n",
       "      <td>gbm_tcga_pan_can_atlas_2018</td>\n",
       "      <td>Glioblastoma Multiforme (TCGA, PanCancer Atlas)</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>29625048,29596782,29622463,29617662,29625055,2...</td>\n",
       "      <td>TCGA, Cell 2018</td>\n",
       "      <td>PUBLIC;PANCAN</td>\n",
       "      <td>0</td>\n",
       "      <td>2023-08-14 08:28:47</td>\n",
       "      <td>True</td>\n",
       "      <td>gbm_tcga_pan_can_atlas_2018</td>\n",
       "      <td>gbm</td>\n",
       "      <td>hg19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecularAlterationType genericAssayType     datatype  \\\n",
       "0           GENERIC_ASSAY     ARMLEVEL_CNA  CATEGORICAL   \n",
       "\n",
       "                                         name  \\\n",
       "0  Putative arm-level copy-number from GISTIC   \n",
       "\n",
       "                                       description  showProfileInAnalysisTab  \\\n",
       "0  Putative arm-level copy-number from GISTIC 2.0.                      True   \n",
       "\n",
       "   patientLevel                        molecularProfileId  \\\n",
       "0         False  gbm_tcga_pan_can_atlas_2018_armlevel_cna   \n",
       "\n",
       "                       studyId  \\\n",
       "0  gbm_tcga_pan_can_atlas_2018   \n",
       "\n",
       "                                        study_name  ... study_publicStudy  \\\n",
       "0  Glioblastoma Multiforme (TCGA, PanCancer Atlas)  ...              True   \n",
       "\n",
       "                                          study_pmid   study_citation  \\\n",
       "0  29625048,29596782,29622463,29617662,29625055,2...  TCGA, Cell 2018   \n",
       "\n",
       "    study_groups study_status     study_importDate study_readPermission  \\\n",
       "0  PUBLIC;PANCAN            0  2023-08-14 08:28:47                 True   \n",
       "\n",
       "                 study_studyId study_cancerTypeId study_referenceGenome  \n",
       "0  gbm_tcga_pan_can_atlas_2018                gbm                  hg19  \n",
       "\n",
       "[1 rows x 21 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = mf.get_molecular_profile(\"gbm_tcga_pan_can_atlas_2018_armlevel_cna\")\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecularAlterationType</th>\n",
       "      <th>datatype</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>showProfileInAnalysisTab</th>\n",
       "      <th>patientLevel</th>\n",
       "      <th>molecularProfileId</th>\n",
       "      <th>studyId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MRNA_EXPRESSION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>mRNA expression (RNA Seq V2 RSEM)</td>\n",
       "      <td>mRNA gene expression (RNA Seq V2 RSEM)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_rna_seq_v2_mrna</td>\n",
       "      <td>acc_tcga</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MRNA_EXPRESSION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>mRNA expression (microarray)</td>\n",
       "      <td>Expression levels for 17155 genes in 590 brca ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>brca_tcga_mrna</td>\n",
       "      <td>brca_tcga</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecularAlterationType    datatype                               name  \\\n",
       "0         MRNA_EXPRESSION  CONTINUOUS  mRNA expression (RNA Seq V2 RSEM)   \n",
       "1         MRNA_EXPRESSION  CONTINUOUS       mRNA expression (microarray)   \n",
       "\n",
       "                                         description  \\\n",
       "0             mRNA gene expression (RNA Seq V2 RSEM)   \n",
       "1  Expression levels for 17155 genes in 590 brca ...   \n",
       "\n",
       "   showProfileInAnalysisTab  patientLevel        molecularProfileId    studyId  \n",
       "0                     False         False  acc_tcga_rna_seq_v2_mrna   acc_tcga  \n",
       "1                     False         False            brca_tcga_mrna  brca_tcga  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3a = mf.fetch_molecular_profiles(molecular_profile_ids=[\"brca_tcga_mrna\", \"acc_tcga_rna_seq_v2_mrna\"])\n",
    "df3a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecularAlterationType</th>\n",
       "      <th>datatype</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>showProfileInAnalysisTab</th>\n",
       "      <th>patientLevel</th>\n",
       "      <th>molecularProfileId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>genericAssayType</th>\n",
       "      <th>pivotThreshold</th>\n",
       "      <th>sortOrder</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PROTEIN_LEVEL</td>\n",
       "      <td>LOG2-VALUE</td>\n",
       "      <td>Protein expression (RPPA)</td>\n",
       "      <td>Protein expression measured by reverse-phase p...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_rppa</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PROTEIN_LEVEL</td>\n",
       "      <td>Z-SCORE</td>\n",
       "      <td>Protein expression z-scores (RPPA)</td>\n",
       "      <td>Protein expression, measured by reverse-phase ...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_rppa_Zscores</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COPY_NUMBER_ALTERATION</td>\n",
       "      <td>DISCRETE</td>\n",
       "      <td>Putative copy-number alterations from GISTIC</td>\n",
       "      <td>Putative copy-number calls on 90 cases determi...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_gistic</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COPY_NUMBER_ALTERATION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>Capped relative linear copy-number values</td>\n",
       "      <td>Capped relative linear copy-number values for ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_linear_CNA</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MUTATION_EXTENDED</td>\n",
       "      <td>MAF</td>\n",
       "      <td>Mutations</td>\n",
       "      <td>Mutation data from whole exome sequencing. Mut...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>acc_tcga_mutations</td>\n",
       "      <td>acc_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecularAlterationType    datatype  \\\n",
       "0           PROTEIN_LEVEL  LOG2-VALUE   \n",
       "1           PROTEIN_LEVEL     Z-SCORE   \n",
       "2  COPY_NUMBER_ALTERATION    DISCRETE   \n",
       "3  COPY_NUMBER_ALTERATION  CONTINUOUS   \n",
       "4       MUTATION_EXTENDED         MAF   \n",
       "\n",
       "                                           name  \\\n",
       "0                     Protein expression (RPPA)   \n",
       "1            Protein expression z-scores (RPPA)   \n",
       "2  Putative copy-number alterations from GISTIC   \n",
       "3     Capped relative linear copy-number values   \n",
       "4                                     Mutations   \n",
       "\n",
       "                                         description  \\\n",
       "0  Protein expression measured by reverse-phase p...   \n",
       "1  Protein expression, measured by reverse-phase ...   \n",
       "2  Putative copy-number calls on 90 cases determi...   \n",
       "3  Capped relative linear copy-number values for ...   \n",
       "4  Mutation data from whole exome sequencing. Mut...   \n",
       "\n",
       "   showProfileInAnalysisTab  patientLevel     molecularProfileId   studyId  \\\n",
       "0                     False         False          acc_tcga_rppa  acc_tcga   \n",
       "1                      True         False  acc_tcga_rppa_Zscores  acc_tcga   \n",
       "2                      True         False        acc_tcga_gistic  acc_tcga   \n",
       "3                     False         False    acc_tcga_linear_CNA  acc_tcga   \n",
       "4                      True         False     acc_tcga_mutations  acc_tcga   \n",
       "\n",
       "  genericAssayType  pivotThreshold sortOrder  \n",
       "0              NaN             NaN       NaN  \n",
       "1              NaN             NaN       NaN  \n",
       "2              NaN             NaN       NaN  \n",
       "3              NaN             NaN       NaN  \n",
       "4              NaN             NaN       NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3b = mf.fetch_molecular_profiles(study_ids=[\"brca_tcga\", \"acc_tcga\"])\n",
    "df3b.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecularAlterationType</th>\n",
       "      <th>datatype</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>showProfileInAnalysisTab</th>\n",
       "      <th>patientLevel</th>\n",
       "      <th>molecularProfileId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>genericAssayType</th>\n",
       "      <th>pivotThreshold</th>\n",
       "      <th>sortOrder</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COPY_NUMBER_ALTERATION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>Capped relative linear copy-number values</td>\n",
       "      <td>Capped relative linear copy-number values for ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>brca_tcga_linear_CNA</td>\n",
       "      <td>brca_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MRNA_EXPRESSION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>mRNA expression (microarray)</td>\n",
       "      <td>Expression levels for 17155 genes in 590 brca ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>brca_tcga_mrna</td>\n",
       "      <td>brca_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MRNA_EXPRESSION</td>\n",
       "      <td>Z-SCORE</td>\n",
       "      <td>mRNA expression z-scores relative to all sampl...</td>\n",
       "      <td>Log-transformed mRNA expression z-scores compa...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>brca_tcga_mrna_median_all_sample_Zscores</td>\n",
       "      <td>brca_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MRNA_EXPRESSION</td>\n",
       "      <td>Z-SCORE</td>\n",
       "      <td>mRNA expression z-scores relative to all sampl...</td>\n",
       "      <td>Log-transformed mRNA expression z-scores compa...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>brca_tcga_rna_seq_v2_mrna_median_all_sample_Zs...</td>\n",
       "      <td>brca_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>METHYLATION</td>\n",
       "      <td>CONTINUOUS</td>\n",
       "      <td>Methylation (HM450)</td>\n",
       "      <td>Methylation (HM450) beta-values for genes in 8...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>brca_tcga_methylation_hm450</td>\n",
       "      <td>brca_tcga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecularAlterationType    datatype  \\\n",
       "0  COPY_NUMBER_ALTERATION  CONTINUOUS   \n",
       "1         MRNA_EXPRESSION  CONTINUOUS   \n",
       "2         MRNA_EXPRESSION     Z-SCORE   \n",
       "3         MRNA_EXPRESSION     Z-SCORE   \n",
       "4             METHYLATION  CONTINUOUS   \n",
       "\n",
       "                                                name  \\\n",
       "0          Capped relative linear copy-number values   \n",
       "1                       mRNA expression (microarray)   \n",
       "2  mRNA expression z-scores relative to all sampl...   \n",
       "3  mRNA expression z-scores relative to all sampl...   \n",
       "4                                Methylation (HM450)   \n",
       "\n",
       "                                         description  \\\n",
       "0  Capped relative linear copy-number values for ...   \n",
       "1  Expression levels for 17155 genes in 590 brca ...   \n",
       "2  Log-transformed mRNA expression z-scores compa...   \n",
       "3  Log-transformed mRNA expression z-scores compa...   \n",
       "4  Methylation (HM450) beta-values for genes in 8...   \n",
       "\n",
       "   showProfileInAnalysisTab  patientLevel  \\\n",
       "0                     False         False   \n",
       "1                     False         False   \n",
       "2                      True         False   \n",
       "3                      True         False   \n",
       "4                     False         False   \n",
       "\n",
       "                                  molecularProfileId    studyId  \\\n",
       "0                               brca_tcga_linear_CNA  brca_tcga   \n",
       "1                                     brca_tcga_mrna  brca_tcga   \n",
       "2           brca_tcga_mrna_median_all_sample_Zscores  brca_tcga   \n",
       "3  brca_tcga_rna_seq_v2_mrna_median_all_sample_Zs...  brca_tcga   \n",
       "4                        brca_tcga_methylation_hm450  brca_tcga   \n",
       "\n",
       "  genericAssayType  pivotThreshold sortOrder  \n",
       "0              NaN             NaN       NaN  \n",
       "1              NaN             NaN       NaN  \n",
       "2              NaN             NaN       NaN  \n",
       "3              NaN             NaN       NaN  \n",
       "4              NaN             NaN       NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df4 = mf.get_all_molecular_profiles_in_study(study_id=\"brca_tcga\", sortBy=\"description\")\n",
    "df4.head(5)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python_tensorflow",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
